cdap-master/src/main/java/co/cask/cdap/master/environment/k8s/MasterTaskExecutorService.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.master.environment.k8s;
+
+import co.cask.cdap.master.spi.environment.MasterEnvironmentContext;
+import co.cask.cdap.master.spi.environment.MasterEnvironmentTask;
+import com.google.common.util.concurrent.AbstractScheduledService;
+import org.apache.twill.common.Threads;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A service for executing {@link MasterEnvironmentTask} periodically.
+ */
+final class MasterTaskExecutorService extends AbstractScheduledService {
```

Comment:
should we extend from `AbstractRetryableScheduledService` instead?

Reply:
Why? We only need simply infinite retry in here.

Reply:
I see.